libkbfs/folder_branch_ops.go

```diff
@@ -2221,7 +2229,7 @@ func (fbo *folderBranchOps) createEntryLocked(
 	}
 
 	// verify we have permission to write
-	md, err := fbo.getMDForWriteLocked(ctx, lState)
+	md, err := fbo.getMDForWriteLockedForName(ctx, lState, name)
```

Comment:
Yeah... I created a new method:
```go
func (fbo *folderBranchOps) canonicalPath(ctx context.Context, dir Node, name string) (string, error)
```
which returns canonical path for a dir node and name.